### DISCUSSION TITLE
regex: propose strict compatibility with Go regexp

### DISCUSSION THREAD

**Discussion by gmlewis** [vlang/v#13301] [Ideas]
One major frustration for adopting V is that its regex standard library is not compatible with [Go's](https://pkg.go.dev/regexp), making porting more difficult than it should be.

My proposal is to port [Go's regexp standard library unit tests](https://go.googlesource.com/go/+/refs/heads/master/src/regexp/) to V and then update V's implementation of regex to pass all unit tests.

Thoughts?

**Comment by JalonSolov** [comment]
That would be good for Go programmers... but what about Java, Perl, PHP, etc., etc., etc.?

**Comment by gmlewis** [comment]
> That would be good for Go programmers... but what about Java, Perl, PHP, etc., etc., etc.?

According to: https://pkg.go.dev/regexp

> The syntax of the regular expressions accepted is the same general syntax used by Perl, Python, and other languages. More precisely, it is the syntax accepted by RE2 and described at https://golang.org/s/re2syntax, except for \C.

which I believe makes it one of the most widely used syntaxes for regexps.
